lib/server/formations.ts

```diff
@@ -189,3 +189,78 @@ export async function searchFormations(
 
   return formations;
 }
+
+async function _mostPopularFormations(limit: number): Promise<FormationData[]> {
+  const { userId } = auth();
+  let queryResponse;
+
+  if (userId) {
+    queryResponse = await turso.execute({
+      sql: `
+        SELECT
+            f.*,
+            COUNT(v.id) AS vote_count,
+            CASE
+                WHEN v2.id IS NOT NULL THEN 1
+                ELSE 0
+            END AS currentUserLiked
+        FROM
+            formations f
+        LEFT JOIN
+            votes v
+        ON
+            f.id = v.formation_id
+        LEFT JOIN
+            votes v2
+        ON
+            f.id = v2.formation_id
+        AND
+            v2.user_id = (:userId)
+        GROUP BY
+            f.id
+        ORDER BY
+            vote_count DESC
+        LIMIT (:limit);
+      `,
+      args: { limit, userId },
+    });
+  } else {
+    queryResponse = await turso.execute({
+      sql: `
+        SELECT
+            f.*,
+            COUNT(v.id) AS vote_count
+        FROM
+            formations f
+        LEFT JOIN
+            votes v
+        ON
+            f.id = v.formation_id
+        WHERE
+            f.name LIKE (:q) OR f.formation LIKE (:q)
```

Comment:
Getting an error running this locally - any idea?
<img width="960" alt="Screenshot 2024-07-16 at 08 49 35" src="https://github.com/user-attachments/assets/1811c3fd-1c8e-40c1-b07f-d06d07c05cbb">

Reply:
This is a good point. I missed this. `q` is never passed in via args.

Reply:
I’ll fix, forgot to test not logged in